README.md

<!-- README.md is generated from README.Rmd. Please edit that file. -->

# pixarfilms <img src="man/figures/logo.png" align="right" alt="" width="120" />

<!-- badges: start -->

[![CRAN
status](https://www.r-pkg.org/badges/version/pixarfilms)](https://CRAN.R-project.org/package=pixarfilms)
[![R build
status](https://github.com/erictleung/pixarfilms/workflows/R-CMD-check/badge.svg)](https://github.com/erictleung/pixarfilms/actions)
[![downloads](https://cranlogs.r-pkg.org/badges/pixarfilms)](https://cran.r-project.org/package=pixarfilms)
[![total\_downloads](https://cranlogs.r-pkg.org/badges/grand-total/pixarfilms)](https://cran.r-project.org/package=pixarfilms)
[![License
CC0](https://img.shields.io/cran/l/pixarfilms)](https://img.shields.io/cran/l/pixarfilms)
<!-- badges: end -->

> R data package to explore Pixar films, the people, and reception data

This package contains six data sets provided mostly in part by
[Wikipedia](https://en.wikipedia.org/wiki/List_of_Pixar_films).

-   `pixar_films` - released and upcoming films
-   `pixar_people` - main people involved in creating films
-   `genres` - movie genres for each film
-   `box_office` - box office reception and budget information
-   `public_response` - critical and public response
-   `academy` - academy awards and nominations

Feel free to [explore the data
immediately](https://pixarfilms-datasette.herokuapp.com/pixarfilms) in
your web browser using [datasette](https://github.com/simonw/datasette)
and SQL.

## Installation

Install `pixarfilms` from CRAN:

``` r
install.packages("pixarfilms")
```

Or you can install the development version of `pixarfilms` from GitHub
with:

``` r
remotes::install_github("erictleung/pixarfilms")
```

## Example

``` r
library(pixarfilms)
pixar_films
#> # A tibble: 27 x 5
#>    number film            release_date run_time film_rating
#>    <chr>  <chr>           <date>          <dbl> <chr>      
#>  1 1      Toy Story       1995-11-22         81 G          
#>  2 2      A Bug's Life    1998-11-25         95 G          
#>  3 3      Toy Story 2     1999-11-24         92 G          
#>  4 4      Monsters, Inc.  2001-11-02         92 G          
#>  5 5      Finding Nemo    2003-05-30        100 G          
#>  6 6      The Incredibles 2004-11-05        115 PG         
#>  7 7      Cars            2006-06-09        117 G          
#>  8 8      Ratatouille     2007-06-29        111 G          
#>  9 9      WALL-E          2008-06-27         98 G          
#> 10 10     Up              2009-05-29         96 PG         
#> # ... with 17 more rows
```

## Documentation

You can find information about the data sets and more
[here](https://erictleung.com/pixarfilms/). The official CRAN release
page can be found [here](https://cran.r-project.org/package=pixarfilms).

## Data

This data here within is not constrained to exploring just within R.

Here are other accessible means of using this data.

-   Through [your
    browser](https://pixarfilms-datasette.herokuapp.com/pixarfilms)
    using datasette and a SQLite database (located at
    `/data-raw/pixarfilms.db`).
-   Through CSV files found within the `data-raw/` directory.

Here are direct links to each data set.

    https://raw.githubusercontent.com/erictleung/pixarfilms/master/data-raw/academy.csv
    https://raw.githubusercontent.com/erictleung/pixarfilms/master/data-raw/box_office.csv
    https://raw.githubusercontent.com/erictleung/pixarfilms/master/data-raw/genres.csv
    https://raw.githubusercontent.com/erictleung/pixarfilms/master/data-raw/pixar_films.csv
    https://raw.githubusercontent.com/erictleung/pixarfilms/master/data-raw/pixar_people.csv
    https://raw.githubusercontent.com/erictleung/pixarfilms/master/data-raw/public_response.csv

There’s also a `datapackage.json` file (located at
`/data-raw/datapackage.json`) to be a computer-readable data dictionary
describing the contents of each data file as described in [the data
package
specifications](https://specs.frictionlessdata.io/data-package/).

## Feedback

If you have any feedback or suggestions on other data that can be added,
please file an issue
[here](https://github.com/erictleung/pixarfilms/issues).

## Code of Conduct

Please note that the {pixarfilms} project is released with a
[Contributor Code of
Conduct](https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html).
By contributing to this project, you agree to abide by its terms.

## Acknowledgments

-   [Wikipedia](https://www.wikipedia.org)
-   [OMDb API](https://www.omdbapi.com/)
-   [babynames](https://github.com/hadley/babynames) (for inspiration)


NEWS.md
# pixarfilms 0.2.1

* Fix for R CMD check change
* Tidy DESCRIPTION film
* Add words to ignore list when spell checking

# pixarfilms 0.2.0

* Add vignette for Pixar film ratings
* Add to suggested packages for vignettes
* Update to HTTPS for README badges

# pixarfilms 0.1.1

* Rename `rating` to `film_rating` to be more clear on meaning
* Update test data based on recent data updates
* Re-knit README for updated data

# pixarfilms 0.1.0

* Update awards for Soul and Onward
* Update people involved in films
* Add film run times and movie ratings to `pixar_films` data
* Add `stringr` library to raw data processing

# pixarfilms 0.0.1

* Added a `NEWS.md` file to track changes to the package.
* Scrape data from Wikipedia and wrangle data into package
* Create `pixarfilms` package


tests/testthat/test-data.R
test_that("Pixar films head and tail", {
  old <- options(tibble.print_min = 20)
  on.exit(options(old))

  # convenience function for getting first and last 10 rows
  # source:
  # https://github.com/hadley/babynames/blob/master/tests/testthat/test-data.R
  first_last <- function(x) {
    n <- nrow(x)
    if (n >= 20) {
      x[c(1:10, (n - 9):n), ]
    } else {
      return(x)
    }
  }

  expect_snapshot(first_last(pixar_films))
  expect_snapshot(first_last(pixar_people))
  expect_snapshot(first_last(genres))

  # Wide data frames truncate large numbers in inconsistent ways
  expect_snapshot(first_last(box_office[, c("film", "budget")]))
  box_office_cols <- c("film", "box_office_us_canada", "box_office_other",
                       "box_office_worldwide")
  expect_snapshot(first_last(box_office[, box_office_cols]))

  expect_snapshot(first_last(public_response))
  expect_snapshot(first_last(academy))
})


R/data.R
#' @importFrom tibble tibble
NULL


#' Pixar films
#'
#' A data set containing Pixar films, their release order, and release date.
#'
#' @format A data frame with 27 rows and 3 variables:
#' \describe{
#'   \item{number}{order of release}
#'   \item{film}{name of film}
#'   \item{release_date}{date film premiered}
#'   \item{run_time}{film length in minutes}
#'   \item{film_rating}{rating based on Motion Picture Association (MPA) film
#'   rating system}
#' }
#' @source \url{https://en.wikipedia.org/wiki/List_of_Pixar_films}
#' @source \url{https://www.omdbapi.com}
#' @examples
#' pixar_films
"pixar_films"


#' People behind Pixar
#'
#' A data set containing the main people involved in each Pixar film.
#'
#' This data set is put into a tidy format, where each row is a film-role data
#' point. Each film can have multiple individuals per role. For example, Toy
#' Story has four screenwriters.
#'
#' @format A data frame with 228 rows and 3 variables:
#' \describe{
#'   \item{film}{name of film}
#'   \item{role_type}{one of five roles: Director, Musician, Producer,
#'     Screenwriter, Storywriter}
#'   \item{name}{individual's name}
#' }
#' @source \url{https://en.wikipedia.org/wiki/List_of_Pixar_films}
#' @examples
#' pixar_people
"pixar_people"


#' Genres describing Pixar films
#'
#' A data set with the different genre categories for each Pixar film.
#'
#' This data set is put into a tidy format, where each row is a film-genre data
#' point. Each film can have multiple genres. For example, Toy Story is
#' categorized into five genres.
#'
#' @format A data frame with 128 rows and 2 variables:
#' \describe{
#'   \item{film}{name of film}
#'   \item{genre}{genre film is categorized into}
#' }
#' @source \url{https://www.omdbapi.com/}
#' @examples
#' genres
"genres"


#' Box office reception numbers
#'
#' A data set with financial and box office gross numbers for each film.
#'
#' @format A data frame with 24 rows and 5 variables:
#' \describe{
#'   \item{film}{name of film}
#'   \item{budget}{movie budget in U.S. dollars}
#'   \item{box_office_us_canada}{box office gross amount in U.S. dollars for
#'     U.S. and Canada}
#'   \item{box_office_other}{box office gross amount in U.S. dollars for other
#'     territories}
#'   \item{box_office_worldwide}{box office gross amount in U.S. dollars
#'     worldwide}
#' }
#' @source \url{https://en.wikipedia.org/wiki/List_of_Pixar_films}
#' @examples
#' box_office
"box_office"


#' Critical and public response
#'
#' A data set with scores of critical and public response.
#'
#' @format A data frame with 23 rows and 5 variables:
#' \describe{
#'   \item{film}{name of film}
#'   \item{rotten_tomatoes}{score from the American review-aggregation website
#'     Rotten Tomatoes; scored out of 100}
#'   \item{metacritic}{score from Metacritic where scores are weighted average
#'     of reviews; scored out of 100}
#'   \item{cinema_score}{score from market research firm CinemaScore; scored by
#'     grades A, B, C, D, and F}
#'   \item{critics_choice}{score from Critics' Choice Movie Awards presented by
#'     the American-Canadian Critics Choice Association (CCA); scored out of
#'     100}
#' }
#' @source \url{https://en.wikipedia.org/wiki/List_of_Pixar_films}
#' @examples
#' public_response
"public_response"


#' Pixar Academy awards and nominations
#'
#' A data set with the awards and nominations of the Academy Awards, popularly
#' known as the Oscars, each Pixar film has won.
#'
#' This data set is put into a tidy format, where each row is a film-award data
#' point.
#'
#' Some films did not qualify for categories so there is no data for that film.
#'
#' @format A data frame with 72 rows and 3 variables:
#' \describe{
#'   \item{film}{name of film}
#'   \item{award_type}{name of award}
#'   \item{status}{status of award}
#' }
#' @source \url{https://en.wikipedia.org/wiki/List_of_Pixar_films}
#' @examples
#' academy
"academy"
